server/api/auth.js

```diff
@@ -105,6 +107,47 @@ router.post("auth.info", auth(), async (ctx) => {
   const user = ctx.state.user;
   const team = await Team.findByPk(user.teamId);
 
+  const authenticationProvider = await AuthenticationProvider.findOne({
+    where: {
+      teamId: team.id,
+    },
+  });
+
+  const userAuthentication = await UserAuthentication.findOne({
+    where: {
+      userId: user.id,
+      authenticationProviderId: authenticationProvider.id,
+    },
+  });
+
+  if (!userAuthentication.accessToken || !userAuthentication.refreshToken) {
+    throw new AuthenticationError("Tokens are not provided");
+  }
+
+  if (authenticationProvider.name === "slack") {
+    const data = await Slack.post("auth.test", {
+      token: userAuthentication.accessToken,
+    });
+
+    if (!data.ok) {
+      const data = await Slack.postUrlEncodedForm("oauth.v2.access", {
+        client_id: process.env.SLACK_KEY,
+        client_secret: process.env.SLACK_SECRET,
+        grant_type: "refresh_token",
+        refresh_token: userAuthentication.refreshToken,
+      });
+
+      if (!data.ok) {
+        throw new AuthenticationError(data.error);
+      }
```

Comment:
Here is particular, these are the [possible errors](https://api.slack.com/methods/oauth.v2.access#errors), and I am not sure how to go about handling them. 

Right now, Giving `404` will do no good because the user will always get stuck here when they retry.

Should we send something like `contact the team`?